service/src/main/kotlin/fi/espoo/evaka/placement/PlacementController.kt

```diff
@@ -221,6 +225,54 @@ class PlacementController(
                         )
                         .also {
                             generateAbsencesFromIrregularDailyServiceTimes(tx, now, body.childId)
+
+                            // Only proceed with future absence and reservation deletion if
+                            // placements range is in future
+                            if (body.endDate.isAfter(now.toLocalDate())) {
+                                val range =
+                                    DateRange(
+                                        when (body.startDate.isAfter(now.toLocalDate())) {
+                                            true -> body.startDate
+                                            false -> now.toLocalDate().plusDays(1)
+                                        },
+                                        body.endDate
+                                    )
+                                // Delete future absences if new absence category is specific
+                                if (
+                                    !body.type
+                                        .absenceCategories()
+                                        .containsAll(
+                                            setOf(
+                                                AbsenceCategory.BILLABLE,
+                                                AbsenceCategory.NONBILLABLE
+                                            )
+                                        )
+                                ) {
+                                    when (body.type.absenceCategories().first()) {
+                                        AbsenceCategory.BILLABLE ->
+                                            deleteFutureNonGeneratedAbsencesByCategoryInRange(
+                                                tx,
+                                                clock,
+                                                body.childId,
+                                                range,
+                                                AbsenceCategory.NONBILLABLE
+                                            )
+                                        AbsenceCategory.NONBILLABLE ->
+                                            deleteFutureNonGeneratedAbsencesByCategoryInRange(
+                                                tx,
+                                                clock,
+                                                body.childId,
+                                                range,
+                                                AbsenceCategory.BILLABLE
```

Comment:
You should put the `when` in the function argument position instead of copy-pasting the whole function call:

```kotlin
deleteFutureNonGeneratedAbsencesByCategoryInRange(
    tx,
    clock,
    body.childId,
    range,
    listOf(when (body.type.absenceCategories().single()) {
        AbsenceCategory.BILLABLE -> AbsenceCategory.NONBILLABLE
        AbsenceCategory.NONBILLABLE -> AbsenceCategory.BILLABLE
    })
)
```